### ISSUE TITLE
Error: Could not load file or assembly Roslyn.Compilers.CSharp.dll - running on mapped network drive

### ISSUE THREAD

**Issue by Cyberlane** [cake-build/cake#861]
### What You Are Seeing?
```
Error: Could not load file or assembly 'file:///Z:\Projects\SimpleTest\tools\cake\Roslyn.Compilers.CSharp.dll' or one of it's dependencies.
```

### What is Expected?
A regular build.

### What version of Cake are you using?
The folder suggests `0.11.0`

### Are you running on a 32 or 64 bit system?
64 bit

### What environment are you running on?  Windows? Linux? Mac?
I have a Macbook running Mac, and my git repo lives on the Mac.
I am running cake within a Windows VM, which has access to the git repo via a network mapped drive (feature given by Parallels).

### Are you running on a CI Server?  If so, which one?
Not yet, just regular powershell command line.

### How Did You Get This To Happen? (Steps to Reproduce)

`Z:\` is a network mapped drive to a folder on my Macbook.
Within the Windows machine I run this command

```
./tools/Cake/Cake.exe -Verbosity=Diagnostic
```

### Output Log

Log with the specific error
https://gist.github.com/Cyberlane/9a869e0371ddeca51de3af108d45c63b

Verbose Diagnostic output log
https://gist.github.com/Cyberlane/a835480fb749cf2c3352f02883b07169

**Comment by gep13** [comment]
@Cyberlane have you linked to the correct Gist?  I am not seeing the error message that you describe within that Gist.

**Comment by devlead** [comment]
I'm guessing you're Windows has placed the network mapped drive under intranet trust or worse, denying it to run arbitrary assemblies. Are you using Windows 7?

You could try changing trust using caspol.exe I.e.
`CasPol -pp off -machine -addgroup 1.2 -url file:///z:/Projects/SimpleTest/* FullTrust`
It's usually located in the framework runtime folder.

You could also try running it under i.e. C:\Temp to prove this theory, if it runs there it's likely a local trust issue.

**Comment by Cyberlane** [comment]
@gep13 sorry about that, the error shows up in the regular log whilst I pasted a verbosity diagnostic log. I have pasted in both links now.

@devlead Windows 10 :)
I will try your caspol trick shortly and report back my findings.

**Comment by Cyberlane** [comment]
@devlead I get access denied to registry when trying to run CasPol in a regular terminal, however I cannot run it in an admin console as the admin console does not see the network mapped drive.

**Comment by devlead** [comment]
@Cyberlane that's "correct" behavior, you'll need to map the drive as administrator, user / admin by design can't see each others network resources.
(if ju run command prompt as non-admin and type `net use` you'll see how Z: is mapped, then you in an admin prompt can use the share path and type i.e. `net use Z: \\computername\sharename`)

But before going to deep on the fulltrust path as it could be something else, if you just clone `https://github.com/cake-build/example.git` to i.e. `c:\temp` on the windows machine and then from PowerShell cd into the `c:\temp\example` folder and run `build.ps1` do you see the same issue or does it successfully compile?

**Comment by Cyberlane** [comment]
@devlead my apologies for taking so long to get back to this, it became a lower priority due to other issues we had at work. Now that I have more time to look at this, I have returned with some fresh eyes.

After I added the `net use...` command, I no longer receive that error, I do get a different one but I believe that's completely unrelated. I will close the issue now, as you have helped me solve the problem! 👍
